pkg/pgmodel/common/extension/extension.go

```diff
@@ -132,9 +132,14 @@ func MigrateExtension(conn *pgx.Conn, extName string, extSchemaName string, vali
 		return fmt.Errorf("Could not get the installed extension version: %w", err)
 	}
 
-	newVersion, ok := getNewExtensionVersion(extName, availableVersions, defaultVersion, validRange, isInstalled, currentVersion)
+	if isInstalled && !upgradeExt {
+		log.Info("msg", "skipping "+extName+" extension upgrade as upgrade extension is disabled. The current extension version is "+currentVersion.String())
```

Comment:
Important: Sorry missed this before. We still have to validate the extension version if not upgrading
```suggestion
		log.Info("msg", "skipping "+extName+" extension upgrade as upgrade extension is disabled. The current extension version is "+currentVersion.String())
		if !validRange(currentVersion) {
		     return fmt.Errorf("The %v extension is not installed at the right version and upgrades are disabled, need version: %v and the installed version is %s", extName, rangeString, currentVersion)
		}
```